scripts/WFCORE-5261-common-conf.adoc

```diff
@@ -0,0 +1,102 @@
+= Add Common Script Configuration File
+:author:            James R. Perkins
+:email:             jperkins@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+There is a desire to have a common script configuration file, `common.conf`. This will be called before the standalone
+or domain script configuration file. The other scripts will also invoke the common script configuration file. It allows
+things like common environment variables to be set by users regardless of the script being used.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-5261[WFCORE-5261]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1251[EAP7-1251]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* WildFly Core
+* WildFly
+* Any project or component which invokes script files
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [ ] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* The common script should be invoked before the standard script configuration files.
+* This will need to work on all supported platforms.
+** `common.conf`
+** `common.conf.bat`
+** `common.conf.ps1`
+* The environment variable `COMMON_CONF` can be used to override the path to the file.
+** By default, `$JBOSS_HOME/bin/common.conf` will be used.
+* All scripts in the `$JBOSS_HOME/bin` directory will invoke this script configuration file.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+* The `common.conf`, `common.conf.bat` and `common.conf.ps1` files do not need to be present in default distribution.
+  Documentation will be added to indicate these files can be created. However, we should not add these files by default
+  as we shouldn't add more clutter to the standard bin directory.
```

Comment:
Will runtime scripts take into account this custom conf as well?

Reply:
If I'm understanding you correctly yes. All scripts will invoke the `common.conf`, `common.conf.bat` or `common.conf.ps1` if they exist. By all scripts I mean the ones in the `$JBOSS_HOME/bin` directory. Looking at it we may need to do some work in WildFly as well as there are some scripts there that do not exist in WildFly Core.